pyterrier/pipelines.py

```diff
@@ -365,6 +388,14 @@ def _apply_round(measure, value):
     elif len(names) != len(retr_systems):
         raise ValueError("names should be the same length as retr_systems")
 
+    # validate save_dir and resulting filenames
+    if save_dir is not None:
+        if not os.path.exists(save_dir):
+            raise ValueError("save_dir %s does not exist" % save_dir)
+        for n in names:
+            if '/' in n:
```

Comment:
This turns out to be a little more complicated for Windows, since there are more reserved path characters: https://stackoverflow.com/questions/1033424/how-to-remove-bad-path-characters-in-python

Reply:
Perhaps I extract a pt.io.ok_filename() method?